internal/venice-common/src/main/java/com/linkedin/venice/utils/Utils.java

```diff
@@ -313,11 +315,28 @@ public static Map<String, String> parseCommaSeparatedStringMapFromString(String
     try {
       Map<String, String> map = new HashMap<>();
       if (!value.isEmpty()) {
-        Arrays.stream(value.split(",")).map(s -> s.split("=")).forEach(strings -> map.put(strings[0], strings[1]));
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode rootNode = objectMapper.readTree(value);
+
+        rootNode.fieldNames().forEachRemaining(key -> {
+          JsonNode childNode = rootNode.get(key);
+          if (childNode.isObject()) {
+            map.put(key, childNode.toString());
+          } else {
+            map.put(key, childNode.asText());
+          }
+        });
```

Comment:
Instead of parsing this JSON string manually, would it work if we use:
```Map<String, String> map = mapper.readValue(value, new TypeReference<Map<String, String>>() {});``` ?